### ISSUE TITLE
KEYLEN_PART_KEY appears unused

### ISSUE THREAD

**Issue by hardenedapple** [neovim/neovim#6831]
This macro is only expanded in `vgetorpeek()`, it is used to compare `keylen` against when deciding whether & how long to wait for.

It's value is `-1`, so that normal searching for a map doesn't set `keylen` to this value.
The only time `keylen` **is** set to this value is after a partial match against `pastetoggle` or the `ui_toggle` key.

When looking for a match against these byte sequences, the condition `mp == NULL` holds (known because the code looking for the match is inside an `if()` clause.

In the case that a partial match for either of these values is found and `keylen` is set to `KEYLEN_PART_KEY`,  then the next `if()` clause will be executed (because the test
```c
if ((mp == NULL || ... ) && keylen != KEYLEN_PART_MAP)
```
will pass).

This clause will `break;` from the `for(;;)` loop looking for characters after having set `c` to a positive, non-zero number.
Hence the enclosing `do {} while()` loop will exit and the rest of the function will proceed.
After the `do {} while ()` loop, there is no use of the local `keylen` variable, and there is no use of the `KEYLEN_PART_KEY` macro outside of the `for(;;)` loop.

To summarise:
- When `keylen` is compared against `KEYLEN_PART_KEY` we know that it can't be equal
- When `keylen` is set to `KEYLEN_PART_KEY` the `for(;;)` loop where `keylen` is used is broken out of and the function returns soon after, throwing that variable away.

**Comment by aktau** [comment]
Seems like reasonably complex control flow, but I'm still wondering why coverity or clang static-analyzer (which version are we running now?) didn't catch this. Especially clang seems to have good control flow annotations...

**Comment by justinmk** [comment]
This would mean that `p_ttimeout` is also unused.

**Comment by zeertzjq** [comment]
It is now used.
